packages/fields/src/fields/title/index.ts

```
@@ -18,7 +18,7 @@ const titleField: Field< CommonPost > = {
 	placeholder: __( 'No title' ),
 	getValue: ( { item } ) => getItemTitle( item ),
 	render: TitleView,
-	enableHiding: false,
+	enableHiding: true,
```

I understand we want to display actions in the media field when title is not present. But, do we also want the title field to be hidable in the existing views (site editor's pages)? Not anything against it, just double-checking.

---

> But, do we also want the title field to be hidable in the existing views

Yes, it'll be a key part of building out [a new media library](https://github.com/WordPress/gutenberg/issues/55238) that's powered by data views, as we'll need to be able to hide the title for attachment post types in a grid view.

---

Yeah, that sounds a good experience for the media library, but can this be addressed differently without affecting existing views? For example, with this view config for the media library:

```js
const view = {
  type: 'grid',
  // titleField not provided
  mediaField: '...',
  fields: [ /**/ ]
};
````

That way, we wouldn't enable hiding the `title` in the site editor's page.

---

> but can this be addressed differently without affecting existing views?

No, I don't think so. For the media use case, we still want the title field to exist, as it's needed for accessibility (i.e. the label provided to the selection checkbox). So it's important that we can _hide_ the title field, without excluding it altogether from the view.

Also, I think it should still be possible for folks to show the title field if they want to, with it defaulted to hidden.

To mitigate the issue for existing views, that's one of the reasons I was proposing making it so that if any locked fields are present, we can't hide the last one that's visible. So we give users more flexibility, but add a guardrail so that they can't get to an unusable state where no fields at all are visible.

---

> That way, we wouldn't enable hiding the title in the site editor's page.

Oh, one detail to mention here is that this PR doesn't affect the pages list in the site editor as that uses a different field ([pageTitleField](https://github.com/WordPress/gutenberg/blob/95e8e91a3d2c0d274336ebf8342a98419728c4cd/packages/fields/src/fields/page-title/index.ts#L14)), rather than the generic `titleField` which is used as a fallback here: https://github.com/WordPress/gutenberg/blob/dd70c03c9eae9fb0806e27cd390227a01f428ad3/packages/editor/src/dataviews/store/private-actions.ts#L192

---

I think it makes sense for the base title to default to enable hiding. If fields can usually be hidden, _disabling_ hiding is a decision which should probably be made at the consumer level. Both Pages and Templates have their own title fields which disable hiding, so nothing will change in the actual site editor.

---

> Oh, one detail to mention here is that this PR doesn't affect the pages list in the site editor as that uses a different field

Oh, nice. My worry was about affecting existing screens 👍 

Little thing (default value is `true`):

```suggestion
```

> disabling hiding is a decision which should probably be made at the consumer level

I think of `wordpress/fields` as consumer-level code! It's just a different package, but I see it as an opinionated fields bundle for WordPress entities that we expect to use in our screens. While any screen can modify the field, in this bundle, it's best to have good defaults.

Thanks to you both for the context. I now understand this change better.

---

Actually, this field is used in the post list experiment ("Gutenberg > Experiments > enable for posts", then visit "Gutenberg > Posts"). Now that I've looked more into how fields are organized, would this work for you instead?

- Leave the existing `title` field as it is.
- **When** it comes to use the title field in the media library, revisit this. We could/may need to add a `mediaTitleField` that has specific field configs. I understand we don't need that now, but this change was more of a future thing, right?

This would prevent making the field hidable for any CPT entity, for example.

In any case, if you still think it's best to carry on with this change, I'd suggest at least adding a changelog entry in the fields package so consumers are aware of this.

---

Thanks for the continued discussion here, I do think it's really good for us to think through these nuances!

My take on it is that the base title field should be the most flexible, with each particular instance that inherits from it being more opinionated. Ideally, I don't think we'd create a separate title field for media unless we have other requirements for how the title is displayed than what's included in the base title field. Or alternately, we might have `postTitle` in addition to `mediaTitle`, keeping `titleField` as the generic / base field.

For now, I'm pretty sold on enabling hiding on this field, so I've added a changelog entry for it. As we explore follow-ups to tackle the UX issues with hiding fields, I'm very happy to revisit this, though!

Once this PR lands, let's keep the discussion going on the linked issue: https://github.com/WordPress/gutenberg/issues/71078

> Little thing (default value is true):

In this case, due to the discussion surrounding hiding, I actually prefer keeping the value explicit here so that if someone looks to make a change to it, they'll see that it was added here intentionally.